Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs

```diff
@@ -29,18 +29,26 @@ private void OnShot(EntityUid uid, WeaponRandomExplodeComponent component, ShotA
             if (_random.Prob(component.explosionChance))
             {
                 var intensity = 1;
+                var reduction = 1; //#funkystation type shit
+                if (component.reduction != null)
+                {
+                    reduction = Convert.ToInt32(component.reduction);
+                }
                 if (component.multiplyByCharge > 0)
                 {
-                    intensity = Convert.ToInt32(component.multiplyByCharge * (battery.CurrentCharge / 100));
+                    intensity = Convert.ToInt32(component.multiplyByCharge * (battery.CurrentCharge / (100 * reduction)));
                 }
 
                 _explosionSystem.QueueExplosion(
                     (EntityUid) uid,
                     typeId: "Default",
                     totalIntensity: intensity,
-                    slope: 5,
-                    maxTileIntensity: 10);
-                QueueDel(uid);
+                    slope: (5 / reduction),
+                    maxTileIntensity: (10 / reduction));
+                if (component.destroyGun == true) //#funkystation part of comp or something idk go away tay i dont wanna do this shit
```

Comment:
redundant check

Reply:
it checks if its true and deletes the gun if its true how is that redundant fatso